there is no test to define HAVE_GRANTPT_PTY_ISATTY and
_POSIX_SOURCE is app-defined not system
This fix ptys and launching xterm

Upstream-Status: Pending

Suggested By Rich Felker
Signed-off-by: Khem Raj <raj.khem@gmail.com>

--- a/main.c
+++ b/main.c
@@ -2892,7 +2892,7 @@ get_pty(int *pty, char *from GCC_UNUSED)
 	close(opened_tty);
 	opened_tty = -1;
     }
-#elif defined(HAVE_POSIX_OPENPT) && defined(HAVE_PTSNAME) && defined(HAVE_GRANTPT_PTY_ISATTY)
+#elif defined(HAVE_POSIX_OPENPT) && defined(HAVE_PTSNAME)
     if ((*pty = posix_openpt(O_RDWR)) >= 0) {
 	char *name = ptsname(*pty);
 	if (name != 0) {
@@ -4040,7 +4040,7 @@ spawnXTerm(XtermWidget xw, unsigned line
 	    /*
 	     * now in child process
 	     */
-#if defined(_POSIX_SOURCE) || defined(SVR4) || defined(__convex__) || defined(__SCO__) || defined(__QNX__)
+#if defined(_POSIX_VERSION) || defined(SVR4) || defined(__convex__) || defined(__SCO__) || defined(__QNX__)
 	    int pgrp = setsid();	/* variable may not be used... */
 #else
 	    int pgrp = getpid();
